src/EnvJson.php

```
@@ -60,85 +70,178 @@
-        $envDistJsonFile = realpath(sprintf('%s/%s', $dir, str_replace('.json', '.dist.json', $jsonName)));
-        if ($envJsonFile) {
-            return json_decode(file_get_contents($envJsonFile), true, 512, JSON_THROW_ON_ERROR); // @phpstan-ignore-line
+        // Construct paths directly instead of using realpath initially
+        $envJsonFile = sprintf('%s/%s', $dir, $jsonName);
+        $envDistJsonFile = sprintf('%s/%s', $dir, str_replace('.json', '.dist.json', $jsonName));
+
+        // Check env.json first
+        if (file_exists($envJsonFile)) {
+            // Check if it's a directory before trying to read
+            if (is_dir($envJsonFile)) {
+                throw new EnvJsonFileNotReadableException("env file is a directory: {$envJsonFile}");
```

PHPStan (2.0.3)

Instantiated class Koriym\EnvJson\Exception\EnvJsonFileNotReadableException not found.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(class.notFound)